test/e2e/tests/utils.py

```
@@ -0,0 +1,599 @@
+    timeout_seconds: int = 600,
+    sleep_seconds: int = 10,
+) -> Any:
+    last_value: Any = None
+    for last_value in TimeoutSampler(
+        wait_timeout=timeout_seconds,
+        sleep=sleep_seconds,
+        func=condition_callable,
+    ):
+        if last_value:
+            return last_value
+    raise AssertionError(f"Timed out waiting for condition: {description}. Last value: {last_value!r}")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)